### PR TITLE
Set the new DSL-level version number attribute for AST requests

### PR DESCRIPTION
**Pull request by sfc-gh-azwiegincew**
This avoids the needlessly complicated process of dealing with the semantic client version number on the server.

Also, set the proper client version number AND the full Python version.

### CODE REVIEW

File: src/snowflake/snowpark/_internal/ast.py [hunk 1]
```diff
@@ -144,8 +144,9 @@ def flush(self) -> Tuple[str, str]:
     def _init_batch(self):
         self._request_id = uuid.uuid4()  # Generate a new unique ID.
         self._request = proto.Request()
-        self._request.client_version.major = 42
```
Comment (sfc-gh-lspiegelberg) [inline src/snowflake/snowpark/_internal/ast.py:147]: should we remove this? what is the client_version here, the Snowpark client version (i.e., 1.20?)?

File: src/snowflake/snowpark/_internal/ast.py [hunk 2]
```diff
@@ -144,8 +144,9 @@ def flush(self) -> Tuple[str, str]:
...
+        self._request.client_version.major = 42  # TODO
         self._request.client_version.minor = 0
         (major, minor, micro, releaselevel, serial) = sys.version_info
         self._request.client_language.python_language.version.major = major
         self._request.client_language.python_language.version.minor = minor
```
Comment (sfc-gh-lspiegelberg) [inline src/snowflake/snowpark/_internal/ast.py:151]: nit: would prefer to have the MAX_VERSION be publicly exposed with an interface/property.

**Review by sfc-gh-lspiegelberg — APPROVED** [review]
not sure about client_version, if possible - can we fix it in this PR as well? Else, lgtm.

File: src/snowflake/snowpark/_internal/ast.py (hunk 1 above)
Comment (sfc-gh-azwiegincew) [reply src/snowflake/snowpark/_internal/ast.py:147]: I added a TODO here.

**Pull request merged**